### DISCUSSION TITLE
Speed tips for users on Intel Macs with non-nvidia cards?

### DISCUSSION THREAD

**Discussion by Kirkman** [openai/whisper#268] [Q&A]
I have Whisper running on a multicore Mac Pro (2012) under Mojave. Video card is Radeon RX 580.

As far as I understand, Whisper will only use the CPU (in FP32) for such a configuration, and will not make use of the GPU. I've used it several times with the tiny and base models and it works well. It's just very slow.

I came across this [promising thread](https://github.com/openai/whisper/discussions/208) about faster CPU execution. 

Is there anything _else_ I can do to speed it up, or take advantage of the hardware I have? Are there any other forks that might allow Whisper to run Radeon on a pre-MacOS-11 Mac? Or a parallelized version that might make use of multiple cores?

**Comment by ArtyomZemlyak** [comment]
Pytorch should use all CPU cores on default.

## For CPU execution

You can try this [great C++ implementation](https://github.com/ggerganov/whisper.cpp) from @ggerganov.
But I think you already saw it in the mentioned thread.
But you need put in mind, that you will be need working with build options. If CPU have AVX2 instructions - Whisper should run fast.

## For GPU execution

Its hard to run DNN models on Radeon...
Maybe try something like this:

1. Use ONNX Models from https://github.com/zhuzilin/whisper-openvino
2. Then try run on onnxruntime: https://onnxruntime.ai/docs/execution-providers/MIGraphX-ExecutionProvider.html

**Reply by ArtyomZemlyak** [reply]
Or this: https://onnxruntime.ai/docs/execution-providers/ROCm-ExecutionProvider.html

**Comment by glangford** [comment]
FYI from the ROCm FAQ :

> Q: Are there plans to support ROCm on MacOS?
> Ans: There are no plans to support ROCm on MacOS currently. For more details, refer to ROCm installation documentation  at:
> https://docs.amd.com/bundle/ROCm-Installation-Guide-v5.2/

https://docs.amd.com/bundle/ROCm-Installation_FAQ/page/Frequently_Asked_Questions.html
